### PR TITLE
Add New feature: existing_query_df and local parameters

### PR DESCRIPTION
**Pull request by skmanzg**
I know you said it should be the form of an independent function, but I have been focusing on adding new features while preserving existing styles instead of creating another new function. Not to mention there will be many unnecessary duplicated code lines if I did that independently. So, I would like you to review it first and then discuss it later.😅

### CODE REVIEW

**Comment by skmanzg** [comment]
bug founded: openAI shape ERROR, cuda 1 setting. (will fix soon)

File: autorag/data/qacreation/llama_index.py [hunk 1]
```diff
@@ -8,6 +9,9 @@
 
 from autorag.utils.util import process_batch
 
+
+nest_asyncio.apply()
```
Comment (vkehfdl1) [inline autorag/data/qacreation/llama_index.py:13]: Recommend to delete this if you add this for jupyter notebook.
Unwanted nest asyncio at non-jupyter environment.

File: autorag/data/qacreation/llama_index.py [hunk 2]
```diff
@@ -18,6 +22,7 @@ def generate_qa_llama_index(
         question_num_per_content: int = 1,
         max_retries: int = 3,
         batch: int = 4,
+        queries: Optional[List[str]] = None,
```
Comment (vkehfdl1) [inline autorag/data/qacreation/llama_index.py:25]: It is better to separate new function when queries is not None.

**Review by vkehfdl1 — CHANGES_REQUESTED** [review]
1. It will be great you separate this feature to independent function
2. Recommend use chromadb.Client instead of FAISS. 
AutoRAG didn't have FAISS dependency. And for embedding search, it looks better to use chromadb like any other AutoRAG functions.

**Comment by vkehfdl1** [comment]
Plus, you need to make corresponding test code at `tests` folder!

**Comment by skmanzg** [comment]
So, I followed most of your requirements:
1. New feature should be independent✔️
2. New feature use Chroma instead of FAISS✔️
3. add test code....? 🤔 (not... yet)


This is a sample code in the JupyterNotebook below:

```
from autorag.data.qacreation import generate_qa_with_existing_queries, make_qa_with_existing_queries
from llama_index.llms.vllm import Vllm
from llama_index.llms.openai import OpenAI
from dotenv import load_dotenv
import pandas as pd

# if you use jupyter, you must use this.
import nest_asyncio
nest_asyncio.apply()


# Prepare your own dataset! (Eaxamples below)
# corpus_df = pd.read_parquet('corpus.parquet')
# df = pd.read_parquet('q.parquet')



prompt = """
Answer the questions based on the contents.

Questions:
{{query}}

contents:
{{text}}

Response:
"""

# if you use OpenAI, you need this.
# load_dotenv()
# llm = OpenAI(model='gpt-3.5-turbo')

# local model or other model other than OpenAI
llm = Vllm(
    model="your model path",
    # tensor_parallel_size=4,    # this option is not available for all. if it is not work, delete it.
    temperature=0.7,
    max_new_tokens=256,
    vllm_kwargs={
        "swap_space": 1,
        "gpu_memory_utilization": 0.8,
    },
)

qa_df = make_qa_with_existing_queries(
    corpus_df, 
    existing_query_df=df, 
    content_size=100, 
    qa_creation_func=generate_qa_with_existing_queries,
    llm=llm, 
    prompt=prompt,
    local=True,
    top_k=5,
    output_filepath= "qa.parquet",
)
```

I haven't tested till the end in OpenAI envrionment due to a lack of tokens. There might be bugs.

**Comment by vkehfdl1** [comment]
@skmanzg I'll check it and add test case + documentation. Thanks

**Comment by skmanzg** [comment]
> @vkehfdl1 I'll check it and add test case + documentation. Thanks

I've read what you've done. I understand you fix the most to fit in AutoRAG system. but is there a part for saving embedding vector DB from Corpus df? In my case, I often had to test or change some parameters with the same data. Since creating corpus vector DB can takes a lot if its size is big, I think there should be save and load Corpus vector DB system or at least, optionally. I might misunderstand what you've done because I do not know well about overall AutoRAG function structures anyway! ... and if that is the case, could you explain to me how it can save and load? 

Thank you in advance.

**Comment by vkehfdl1** [comment]
@skmanzg 
I agree.
If so, I'll make a chromadb Collection as parameter. 
If you want to use it locally, or temp collection, or remote server Chroma, the whole case will be cover if we get the collection.
(Users have to write their own code that loading chromadb collection, but it is piece of cake)

**Comment by vkehfdl1** [comment]
@bwook00 Review please ^^

**Comment by skmanzg** [comment]
> (Users have to write their own code that loading chromadb collection, but it is piece of cake)
I doubt about it.


@vkehfdl1 so.. I see collection is for loading DB. but there is no way to save DB from this function... right?

**Comment by vkehfdl1** [comment]
All test was passed at the local.
This is made by outside contributor, so cause error at unit test (No OpenAI API Key)

**Review by vkehfdl1 — APPROVED** [review]
@skmanzg Nothing changed on your force-push right?

**Comment by skmanzg** [comment]
> @vkehfdl1 Nothing changed on your force-push right?

Not at all. I just updated the branch with rebase.

**Comment by vkehfdl1** [comment]
@skmanzg I am bypassing auto-testing (due to the OPENAI API KEY) and merge it.

**Pull request merged**